### ISSUE TITLE
stored self consensus state at current height vs past heights

### ISSUE THREAD

**Issue by colin-axner** [cosmos/ibc#432]
The connection handshakes have the following requirement:
`abortTransactionUnless(consensusHeight <= getCurrentHeight())`

How can a chain retrieve its self consensus state at the current height? Shouldn't it still be processing transactions and forming a root?

**Comment by cwgoes** [comment]
Yes, that's true, the retrieval will just fail, the check can be changed to `<` instead of `<=`.

**Comment by gjermundgaraba** [comment]
This issue was mistakenly reopened during some integration work, so closing again now.
